### PR TITLE
[Housekeeping] Bump Go version to 1.22

### LINKED ISSUE #4994: [Housekeeping] Move to go 1.22
### Describe the issue

Go 1.22 was [released](https://go.dev/blog/go1.22) on 2/6/2024. This release mentions a 2-14% performance increase.

### What if we do not do this?

We'll eventually have to move go versions, so tracking this in this issue for now.

### Related component(s)

_No response_

### Are you sure this issue hasn't been raised already?

- [X] Yes

### Have you read the Code of Conduct?

- [X] Yes

### PR DESCRIPTION
**Pull request by lowc1012**
## Tracking issue
Closes #4994

## Check all the applicable boxes 

- [ ] I updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [x] All commits are signed-off.

### CODE REVIEW

**Comment by eapolinario** [comment]
@lowc1012 , are you still going to work on this PR? If so, can you merge master?

File: Dockerfile [hunk 1]
```diff
@@ -2,7 +2,7 @@ ARG FLYTECONSOLE_VERSION=latest
 FROM ghcr.io/flyteorg/flyteconsole:${FLYTECONSOLE_VERSION} AS flyteconsole
 
 
-FROM --platform=${BUILDPLATFORM} golang:1.21.5-bookworm AS flytebuilder
+FROM --platform=${BUILDPLATFORM} golang:1.22.1-bookworm AS flytebuilder
```
[the remark this answers is not in the document]
Comment (lowc1012) [reply Dockerfile]: updated, thanks @ddl-ebrown

**Comment by eapolinario** [comment]
@lowc1012 , building the sandbox is hitting a failure, can you take a look? You can reproduce locally by running `make -C docker/sandbox-bundled build` from the root of the repo.

**Comment by lowc1012** [comment]
> @lowc1012 , building the sandbox is hitting a failure, can you take a look? You can reproduce locally by running `make -C docker/sandbox-bundled build` from the root of the repo.

I executed `make -C docker/sandbox-bundled build` locally, but got this error:
`ERROR: failed to solve: ghcr.io/flyteorg/flyteconsole:latest: failed to resolve source metadata for ghcr.io/flyteorg/flyteconsole:latest: failed to authorize: failed to fetch oauth token: unexpected status from GET request to https://ghcr.io/token?scope=repository%3Aflyteorg%2Fflyteconsole%3Apull&service=ghcr.io: 403 Forbidden`

**Comment by lowc1012** [comment]
@eapolinario thanks, I push a new commit

**Review by pingsutw — APPROVED** [review]

**Review by eapolinario — COMMENTED** [review]
Can you also update https://github.com/flyteorg/flyte/blob/master/boilerplate/flyte/golang_support_tools/go.mod#L3 ?

**Comment by lowc1012** [comment]
> Thank you, this is looking pretty good. Can you also update https://github.com/flyteorg/flyte/blob/master/boilerplate/flyte/golang_support_tools/go.mod#L3 ?

@eapolinario done

**Review by pingsutw — APPROVED** [review]
cc @eapolinario mind taking another look as well?

**Review by eapolinario — APPROVED** [review]
Thank you. This looks great!

**Pull request merged**